File: lab2/1.py
class Complex(real,img):
	def __init__(self, real, img):
		self.real=real
		self.img=img
	def __add__(self, b):
		return Complex(self.real+b.real, self.img+self.img)
	def __sub__(self, b):
		return Complex(self.real-b.real, self.img-self.img)
	def __mul__(self, b):
		return Complex(self.real*b.real-self.img*b.img,self.real*b.img+self.img*b.real)
	def __floatdiv__(self, b):
		m = b.real*b.real + b.img*b.img
		if m==0:
			print("Invalid Input!")
			exit()
		return  Complex((self.real*b.real+self.img*b.img)/m,(self.img*b.real-self.real*b.img)/m)

def printComplex(a):
	print(a.self, '+', a.img,"i")
	
def getInput():
	print("Input real and imaginary part of first complex no: (a + ib)")
	a=float(input())
	b=float(input())
	print("Input real and imaginary part of second complex no: (c + id)")
	c=float(input())
	d=float(input())
	return a,b,c,d

if __name__ == "main":
	a,b,c,d=getInput()
	x = Complex(a,b)
	y = Complex(c,d)
	printComplex(x)
	
	
	

File: lab1/1.cpp
#include<iostream>
using namespace std;



struct c{
	float first;	// real
	float second;	// imaginary
};

int main()
{
	typedef struct c complex;
	// real,  imaginary
	//typedef ii complex;
	complex a,b;
	// input
	cout<<"Input real and imaginary part of first complex no: (a + ib)"<<endl;
	cin>> a.first >> a.second;
	cout<<"Input real and imaginary part of second complex no: (c + id)"<<endl;
	cin>> b.first >> b.second;

	cout<<"Addition: (a+c)+(b+d)i"<<endl;
	cout<<"\t"<< a.first+b.first <<" + ("<< a.second+b.second <<")i"<<endl;
	cout<<"Subtraction: (a-c)+(b-d)i"<<endl;
	cout<<"\t"<< a.first-b.first <<" + ("<< a.second-b.second <<")i"<<endl;
	cout<<"Multiplication: (ac-bd)+(ad+bc)i"<<endl;
	cout<<"\t"<< a.first*b.first-a.second*b.second <<" + ("<< a.first*b.second+a.second*b.first <<")i"<<endl;
	cout<<"Division: (ac+bd)/(c*c+d*d)+(bc-ad)/(c*c+d*d)i"<<endl;
	float m = b.first*b.first + b.second*b.second;
	cout<<"\t"<< (a.first*b.first+a.second*b.second)/m <<" + ("<< (a.second*b.first-a.first*b.second)/m <<")i"<<endl;
	
	return 0;
}